### PR TITLE
Last round of refactorings!

### LINKED ISSUE #391: Reduce complexity of the code
I've recently run `mlint -cyc` (or `checkcode -cyc`) on the current (i.e. 4202e7360d7e42aa24b7668e26c3dad3a7dfc1d5)  `matlab2tikz.m` file to determine the [cyclomatic complexity](http://en.wikipedia.org/wiki/Cyclomatic_complexity). This is a rough estimate of how complicated a function is and it basically measures how many paths there are through a function (e.g. `if`s, `switch/case`es, `return`s typically increase this measure) there are in the function, which again is a rough estimate of how hard the code is to understand, test and maintain.

Ideally, each function should have a complexity of less than about `10`, unless there are good reasons why it should be more complex. Below is a table that shows the cyclomatic complexity of every function we have:

Line | Function                     | Complexity
----:|:-----------------------------|:---------:
723  | `drawAxes`                   | 51         
3740 | `getLegendOpts`              | 37         
2233 | `drawText`                   | 28         
576  | `handleAllChildren`          | 27    
        | *others removed for brevity* |    

As with any metric, the results have to be interpreted with due care and not every value `>10` is by definition bad, but it might be an indication too much is being done in a single function. 
The following list of functions should be re-evaluated to see whether we can make them simpler (e.g. by introducing new functions) or whether the complexity is inherent to what the function is doing and is acceptable:

- [x] `drawAxes`         
- [x] `drawBarseries`         
- [x] `drawHggroup`         
- [x] `drawImage`      
- [x] `drawLine`         
- [x] `drawPatch`        
- [x] `drawQuiverGroup` 
- [x] `drawScatterPlot`         
- [x] `drawSurface`         
- [x] `drawText`                  
- [x] `drawTextbox`   
- [x] `findLegendInformation`
- [x] `getAxisOptions`         
- [x] `getColorbarOptions`         
- [x] `getLegendOpts`         
- [x] `getMarkerOptions`     
- [x] `getPositionOfText`    
- [x] `handleAllChildren`      
- [x] `legendPosition`   
- [x] `makeTable`        
- [x] `matlab2tikz`      
- [x] `matlabTicks2pgfplotsTicks`         
- [x] `parseTexSubstring`
- [x] `patchcolor2xcolor`         
- [x] `prettyPrint`      
- [x] `saveToFile`         
- [x] `surfaceOpts`         
- [x] `surfaceOptsOfSurf`
- [x] `shaderOptsSurfPatch`
- [x] `translateMarker`         

E.g. `translateMarker` is just a big `switch` statement which essentially translates 15 different marker definitions from MATLAB to TikZ. The code is clear to read, so the complexity is not a major issue (not that we cannot reduce its complexity, but it wouldn't help readability that much).

E.g. in `drawText`, the last part converts a 2D/3D from MATLAB to TikZ and checks whether it lies outside the axis. This code is (almost) duplicated for 2D and 3D and hence could be refactored out into another function.

**Note** If you have checked a certain function, please leave a message (in the code if appropriate and here). You can also check the function above to mark it done.

### PR DESCRIPTION
**Pull request by egeerardyn**
The last set of simplifications to fix #391.
Covered by the R2014b headless tests to make sure everything still works.

I also added a link to the FIXME regarding RGB scatter plots to a stackoverflow post by Christian for a way to work around that hiatus in pgfplots and our code.

### CODE REVIEW

**Comment by PeterPablo** [comment]
You on purpose didn't add the function separator comment line `% ==============================================================================` everywhere -- correct?
Looks good! :+1: -- I couldn't spot regressions.

File: src/matlab2tikz.m [hunk 1]
```diff
@@ -5851,6 +5879,16 @@ function printAll(m2t, env, fid)
         end
     end
 end
+function opts = opts_copy(opts_from, name_from, opts, name_to)
```
Comment (PeterPablo) [inline src/matlab2tikz.m]: :+1:

**Comment by PeterPablo** [comment]
No objections -- feel free to merge.

**Comment by okomarov** [comment]
Same

Sent from my iPhone

On 17 Apr 2015, at 16:35, "Peter Pablo" <notifications@github.com<mailto:notifications@github.com>> wrote:


No objections -- feel free to merge.

**Comment by egeerardyn** [comment]
@PeterPablo I indeed purposefully do not insert `%==...==` between those functions are strongly related. If we are ever to move out some parts of the code, they should remain together and this is my way of indicating that.

**Comment by PeterPablo** [comment]
Just what I thought.

**Pull request merged**